lemur/plugins/lemur_gcp/certificates.py

```diff
@@ -0,0 +1,89 @@
+from flask import current_app
+from google.cloud.compute_v1.services import ssl_certificates
+
+from lemur.common.defaults import common_name, issuer, not_before
+from lemur.common.utils import parse_certificate, split_pem
+
+
+def get_name(body):
+    """
+    We need to change the name of the certificate that we are uploading to comply with GCP naming standards.
+    The cert name will follow the convention "ssl-{Cert CN}-{Date Issued}-{Issuer}"
+    """
+    cert = parse_certificate(body)
+    cn = common_name(cert)
+    authority = issuer(cert)
+    issued_on = not_before(cert).date()
+
+    cert_name = f"ssl-{cn}-{authority}-{issued_on}"
+
+    return modify_cert_name_for_gcp(cert_name)
+
+
+def modify_cert_name_for_gcp(cert_name):
+    # Modify the cert name to comply with GCP naming convention
+    gcp_name = cert_name.replace('.', '-')
+    gcp_name = gcp_name.replace('*', "star")
+    gcp_name = gcp_name.lower()
+    gcp_name = gcp_name[:63]
+    gcp_name = gcp_name.rstrip('.*-')
+    return gcp_name
+
+
+def full_ca(body, cert_chain):
+    # in GCP you need to assemble the cert body and the cert chain in the same parameter
+    return f"{body}\n{cert_chain}"
+
+
+def insert_certificate(project_id, ssl_certificate_body, credentials):
+    return ssl_certificates.SslCertificatesClient(credentials=credentials).insert(
+        project=project_id, ssl_certificate_resource=ssl_certificate_body
+    )
+
+
+def fetch_all(project_id, credentials):
+    client = ssl_certificates.SslCertificatesClient(credentials=credentials)
+    certs = []
+    for cert_meta in client.list(project=project_id):
+        try:
+            if cert_meta.type_ != "SELF_MANAGED":
+                continue
+            cert = parse_certificate_meta(cert_meta)
+            if cert:
+                certs.append(cert)
+        except Exception as e:
```

Comment:
Do we want to re-raise an exception in this block? 
Correct me if I'm wrong, but as is, if an error is raised in the execution flow above then it will be caught here, logged, and then execution will continue normally. In this case `fetch_all` will return an empty array (also causing `GCPSourcePlugin.get_certificates` to return an empty array). My worry here is that to user or cli command calling `GCPSourcePlugin.get_certificates` it will look like code executed successfully and there are no certs to be returned, when in reality the code could have failed.